### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by wale-prog**
In this pull request, I have implemented the following:
- Created a class called `Nameable` and implemented a method `correct_name` that raise a `NotImplementedError`
- `Person` class inherits from `Nameable` and implemented the `correct_name` method that returns the name attribute
- Created base decorated class named `decorator` that inherits from `Nameable` 
- Created `CapitalizeDecorator`  class that inherits from the `Decorator` class
- Created `TrimmerDecorator`  class that inherits from the `Decorator` class

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @wale-prog,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Your linters failed

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require './nameable'
```
Comment (rloterh) [inline decorator.rb:3]: - [ ] OPTIONAL
It's recommended to use `require_relative` for classes within a structure and rather use `require` for _gems_ and other packer imports. Kindly endeavor to improve upon this aspect

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,12 @@
 # frozen_string_literal: true
 
+require './nameable'
+require './decorator'
```
Comment (rloterh) [inline person.rb:4]: - [ ] OPTIONAL
It's recommended to use `require_relative` for classes within a structure and rather use `require` for _gems_ and other packer imports. Kindly endeavor to improve upon this aspect

File: trimmer_decoration.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require './decorator'
```
Comment (rloterh) [inline trimmer_decoration.rb:3]: - [ ] OPTIONAL
It's recommended to use `require_relative` for classes within a structure and rather use `require` for _gems_ and other packer imports. Kindly endeavor to improve upon this aspect

File: capitalize_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+require './decorator'
```
Comment (rloterh) [inline capitalize_decorator.rb:3]: - [ ] OPTIONAL
It's recommended to use `require_relative` for classes within a structure and rather use `require` for _gems_ and other packer imports. Kindly endeavor to improve upon this aspect

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Wale,  

Your project is complete, great job🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Highlights 👍
- Decorator pattern properly implemented

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by sole contributor. 
- It's recommended that you add an MIT license, or any other license to your ReadMe.
Implementing the above changes will make your ReadMe appear more professional.
<br>
Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decoration.rb (hunk 3 above)
Comment (wale-prog) [reply trimmer_decoration.rb:3]: Thanks, I'll do that.

**Pull request merged**